c-scape/src/process/exec.rs

```diff
@@ -0,0 +1,190 @@
+use rustix::fd::BorrowedFd;
+use rustix::fs::{AtFlags, Mode, OFlags};
+
+use alloc::vec::Vec;
+use core::ffi::CStr;
+
+use crate::{Errno, set_errno};
+
+use libc::{c_char, c_int};
+
+extern "C" {
+    static mut environ: *mut *mut c_char;
+}
+
+#[no_mangle]
+unsafe extern "C" fn execl(
+    path: *const c_char, 
+    arg: *const c_char,
+    mut argv: ...
+) -> c_int {
+    let mut vec = Vec::new();
+    vec.push(arg);
+
+    loop {
+        let ptr = argv.arg::<*const c_char>();
+        vec.push(ptr);
+        if ptr.is_null() {
+            break;
+        }
+    }
+    
+    execv(path, vec.as_ptr())
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn execle(
```

Comment:
This shouldn't be `pub`. I don't think it breaks anything; I just want to be consistent about how we use symbols.